web/api/queries/zscore_s/zscore_s.c

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+#include "zscore_s.h"
+
+// ----------------------------------------------------------------------------
+// stddev
+
+// this implementation comes from:
+// https://www.johndcook.com/blog/standard_deviation/
+
+struct stats_zscore {
+    long count;
+    calculated_number m_oldM, m_newM, m_oldS, m_newS, value;
+};
+
+void *stats_create_zscore(RRDR *r) {
+    UNUSED (r);
+    return callocz(1, sizeof(struct stats_zscore));
+}
+
+// resets when switches dimensions
+// so, clear everything to restart
+void stats_reset_zscore(RRDR *r, int index) {
+    struct stats_zscore *g = (struct stats_zscore *)r->stats[index].stat_data;
+    g->count = 0;
+}
+
+void stats_free_zscore(RRDR *r, int index) {
+    freez(r->stats[index].stat_data);
+    r->stats[index].stat_data = NULL;
+}
+
+void stats_add_zscore(RRDR *r, calculated_number value, int index) {
+    struct stats_zscore *g = (struct stats_zscore *)r->stats[index].stat_data;
+
+    if(calculated_number_isnumber(value)) {
+        g->count++;
+        g->value = value;
+        // See Knuth TAOCP vol 2, 3rd edition, page 232
+        if (g->count == 1) {
+            g->m_oldM = g->m_newM = value;
+            g->m_oldS = 0.0;
+        }
+        else {
+            g->m_newM = g->m_oldM + (value - g->m_oldM) / g->count;
+            g->m_newS = g->m_oldS + (value - g->m_oldM) * (value - g->m_newM);
```

Comment:
I understand that we have different algorithms to calculate the variance/standard deviation, so why did you choose this algorithm? (ping @andrewm4894 )

Reply:
I was expecting something like this, but the different `mean` do set the variable `m_newS` created a concern.